substrate/client/src/client.rs

```diff
@@ -251,6 +251,13 @@ impl<B, E, Block> Client<B, E, Block> where
 		&self.executor
 	}
 
+	/// Returns the runtime metadata as JSON.
+	pub fn json_metadata(&self, id: &BlockId<Block>) -> error::Result<String> {
+		self.executor.call(id, "json_metadata",&[])
+			.and_then(|r| String::decode(&mut &r.return_data[..])
+					  .ok_or("Metadata decoding failed".into()))
```

Comment:
check style guide on indentation. basically, it's only ever a single tab at a time, never *ever* mix tabs + spaces.